File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* ptr = head;
        ListNode* cur = head;
        for(int i=0;i<n;i++) ptr = ptr->next;
        if(ptr == NULL) return head->next;
        while(ptr->next){
            ptr = ptr->next;
            cur = cur->next;
        }
        cur->next = cur->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  ListNode? removeNthFromEnd(ListNode? head, int n) {
    if(head == null) return head;
    List<ListNode?> list = [];
    ListNode? cur = head;
    while(cur != null){
        list.add(cur);
        cur = cur.next;
    }
    if(list.length - n == 0){
        return head.next;
    }
    list[list.length-n-1]!.next = list[list.length-n]!.next;
    return head;
  }
}